examples/data_preprocessing/wiennerisches_diarium_historical_dataset/data_preprocessing_sterbelisten_wiener_diarum.ipynb

```diff
@@ -0,0 +1,1246 @@
+    "- X Matrix containing the samples (pandas data frame).\n",
+    "\n",
+    "\n",
+    "##### Pipeline\n",
+    "\n",
+    "**1. Read in the html file**\n",
+    "\n",
+    "- Strip of tags and clean the text with regular expressions.\n",
+    "\n",
+    "- Create a list with each sentence as a sample.\n",
+    "\n",
+    "We decided to keep the abbreviations as the expansion would already involve normalisation and we want to try to work with the 'raw' data.\n",
+    "\n",
+    "\n",
+    "**2. Read in already extracted placenames**\n",
+    "- Extract all \\<mark> words from the .html file provided by the ACDH-CH research group and assign an ID.\n",
+    "- Loop through all samples and create lists containing either None if the word is not a place name or the ID of the place name if the word is part of a place name.\n",
+    "\n",
+    "\n",
+    "**3. Building training data**\n",
+    "\n",
+    "**X Matrix**\n",
+    "pandas data frame or list with all  cleaned sentences.\n",
+    "\n",
+    "**T Matrix**\n",
+    "\n",
+    "np.sparse matirx one column 0, one column 1, rows are place names.\n",
+    "\n",
+    "**Z Matrices**\n",
+    "numpy.narray\n",
+    "columns= keywords \n",
+    "rows = words in sentence\n",
+    "\n",
+    "\n",
+    "**5. Building test data** \n",
+    "\n",
+    "manually checked! \n",
+    "There is the 'timemachine_evaluatuin_v1_edited_corrected.jsonl which should contain manually corrected place names but it is not usable for our purposes."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "##### Please check requirements are in the requirements.txt \n",
+    "\n",
+    "The code in this tutorial is also available as a data_preprocessing_wiener_diarum_toponym.py.\n",
+    "\n",
+    "This folder contains: \n",
+    "- data_preprocessing_sterbelisten_wiener_diarum.ipn\n",
+    "- data_preprocessing_wiener_diarum_toponym.py\n",
+    "- requirements.ipynb\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Imports**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "import re\n",
+    "\n",
+    "import pandas as pd\n",
+    "import numpy as np\n",
+    "from typing import List, Tuple, Dict\n",
+    "\n",
+    "from tqdm import tqdm\n",
+    "from joblib import dump\n",
+    "from minio import Minio\n",
+    "client = Minio(\"knodle.cc\", secure=False)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "'../../../data_from_minio/wiener_diarum_toponyms'"
+      ]
+     },
+     "execution_count": 2,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# define the path to the folder where the data will be stored\n",
```

Comment:
From the Pep 8 style Guide for comments: "Comments should be complete sentences. The first word should be capitalized, unless it is an identifier that begins with a lower case letter."